docs/content/guides/dagster-pipes.mdx

```diff
@@ -0,0 +1,78 @@
+---
+title: Dagster Pipes | Dagster Docs
+description: "Dagster Pipes provides a protocol between the orchestration environment (Dagster) and external execution (ex: Databricks) and a toolkit for building implementations of that protocol."
+---
+
+# Dagster Pipes
+
+Dagster Pipes is a toolkit for building integrations between Dagster and external execution environments. It standardizes the process of passing parameters, injecting context information, ingesting logs, and collecting metadata all while remaining agnostic to how remote computations are launched in those environments. This enables the separation of orchestration and business logic in the Dagster ecosystem.
+
+It also smooths the process of incorporating pre-existing code, business logic, and execution environments into Dagster. Using Dagster Pipes, you add just a few lines to your code and it becomes seamlessly invocable by the orchestrator. In turn, you can stream logs and metadata back to Dagster so you can leverage its observability, lineage, cataloging, and debugging capabilities.
+
+---
+
+## Benefits
+
+With Dagster Pipes, you can:
+
+- Incorporate existing code into Dagster without huge refactors
+- Onboard stakeholder teams onto Dagster incrementally
+- Run code in external environments, and:
+  - Easily pass parameters to the code
+  - Stream unstructured logs and structured metadata back to Dagster
+- Separate orchestration and business logic environments
+- Use languages other than Python with Dagster
+
+---
+
+## Limitations
+
+While Dagster Pipes is lightweight and flexible, there are a few limitations to be aware of:
+
+- **Step launchers are currently incompatible.**
+- **I/O managers aren’t currently supported.** If you rely heavily on [I/O managers](/concepts/io-management/io-managers) in remote execution, we recommend using step launchers.
```

Comment:
@yuhan Is this correct? It feels weird to say I/O managers aren't supported for this, so use step launchers, which also aren't supported. I feel like there's something I missed in our original meeting.

Reply:
I think the io manager "incompatibility" means:

Dagster Pipes is not built for processing data in memory, which I/O managers are for. For example, if you'd like to process data in memory in Pyspark, use I/O manager as demonstrated  https://docs.dagster.io/integrations/spark#running-pyspark-code-in-assets; otherwise (in a remote case), use Pipes.

Reply:
🙏